### PR TITLE
T15422 version top level

### LINKED ISSUE #15422: [NFR]: Remove top level classes: Version
- Move `Version` from the top level namespace
- Add it to a new `Support` namespace
- Make it non static

### LINKED ISSUE #15424: [BUG]: Phalcon\Cache - keys need to be strings
For cache operations, keys need to be casted to strings prior to sending them to adapters, to avoid fatal errors.

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: enhancement
* Link to issue: Fixes #15422 #15424 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

`Phalcon\Version` is not moved to `Phalcon\Support\Version`
    - `_getSpecialVersion` and `_getVersion` have been removed (marked deprecated in v4)
    - The class is no longer static; it has to be instantiated first
    - References to `Phalcon\Debug` and the Volt compiler have been adjusted 
Corrected `Phalcon\Cache` to cast keys as strings before sending them to adapters

Thanks

### CODE REVIEW

File: phalcon/Cache.zep [hunk 1]
```diff
@@ -60,6 +60,8 @@ class Cache implements CacheInterface
      */
     public function delete(var key) -> bool
     {
+        let key = (string) key;
```
Comment (Jeckerson) [inline phalcon/Cache.zep:63]: It is possible to change method signature -> `public function delete(string key)`

File: phalcon/Cache.zep [hunk 2]
```diff
@@ -103,6 +105,8 @@ class Cache implements CacheInterface
      */
     public function get(var key, var defaultValue = null) -> var
     {
+        let key = (string) key;
```
Comment (Jeckerson) [inline phalcon/Cache.zep:108]: It is possible to change method signature -> `public function get(string key)`

File: phalcon/Cache.zep [hunk 3]
```diff
@@ -144,6 +148,8 @@ class Cache implements CacheInterface
      */
     public function has(var key) -> bool
     {
+        let key = (string) key;
```
Comment (Jeckerson) [inline phalcon/Cache.zep:151]: `has(string key)`

File: phalcon/Cache.zep [hunk 4]
```diff
@@ -164,6 +170,8 @@ class Cache implements CacheInterface
      */
     public function set(var key, var value, var ttl = null) -> bool
     {
+        let key = (string) key;
```
Comment (Jeckerson) [inline phalcon/Cache.zep:173]: `set(string key ...`

File: phalcon/Cache.zep (hunk 1 above)
Comment (niden) [reply phalcon/Cache.zep:63]: No because PSR uses `mixed` for that.

File: phalcon/Cache.zep (hunk 2 above)
Comment (niden) [reply phalcon/Cache.zep:108]: Same as above

File: phalcon/Cache.zep (hunk 3 above)
Comment (niden) [reply phalcon/Cache.zep:151]: Same

File: phalcon/Cache.zep (hunk 4 above)
Comment (niden) [reply phalcon/Cache.zep:173]: Same

**Review by Jeckerson — APPROVED** [review]

**Pull request merged**